files/image_config/monit/restart_container

```diff
@@ -0,0 +1,127 @@
+        return 2, command_stdout.strip(), err
+
+    return 0, command_stdout.strip(), command_stderr.strip()
+
+
+def check_container_running(container_name):
+    """Check whether a container is in the running state.
+
+    Args:
+        container_name: Name of a container.
+
+    Returns:
+        None.
+    """
+    start_time = time.time()
+    elapsed_time = 0
+    container_state_command = "docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name)
+
+    syslog.syslog(syslog.LOG_INFO, "Issues command '{}' to check running status of container '{}' ..."
+                  .format(container_state_command, container_name))
+
+    while elapsed_time < CONTAINER_RESTART_THRESHOLD_SECS:
+        exit_code, command_stdout, command_stderr = get_command_result(container_state_command)
+        if exit_code == 0 and command_stdout == "true":
+            syslog.syslog(syslog.LOG_INFO, "Container '{}' is restarted successfully and is in the running state."
+                          .format(container_name))
+            return
+
+        time.sleep(CONTAINER_CHECK_INTERVAL_SECS)
+        elapsed_time = time.time() - start_time
+
+    if elapsed_time >= CONTAINER_RESTART_THRESHOLD_SECS:
+        syslog.syslog(syslog.LOG_ERR, "Container '{}' is not running after {} seconds."
+                      .format(container_name, CONTAINER_RESTART_THRESHOLD_SECS))
+
+
+def reset_failed_flag_before_restart(container_name):
+    """Reset the failed status of a service.
+
+    Args:
+        container_name: Name of the service.
+
+    Returns:
+        None
+    """
+    reset_failed_command = "sudo systemctl reset-failed {}.service".format(container_name)
+
+    syslog.syslog(syslog.LOG_INFO, "Issues command '{}' to reset failed status of service '{}' ..."
+                  .format(reset_failed_command, container_name))
+
+    exit_code, command_stdout, command_stderr = get_command_result(reset_failed_command)
+    if exit_code == 0:
+        syslog.syslog(syslog.LOG_INFO, "Succeeded to reset failed status of service '{}.service'."
+                      .format(container_name))
+    else:
+        syslog.syslog(syslog.LOG_ERR, "Failed to reset failed status of service '{}'. Error: {}"
+                      .format(container_name, command_stderr))
+
+
+def restart_container(container_name):
+    """Reset the failed status of a service and then restart the
+    corresponding container.
+
+    Args:
+        container_name: Name of specified container.
+
+    Returns:
+        None.
+    """
+    restart_command = "sudo systemctl restart {}.service".format(container_name)
+
+    reset_failed_flag_before_restart(container_name)
+
+    syslog.syslog(syslog.LOG_INFO, "Issues command '{}' to restart container '{}' ..."
+                  .format(restart_command, container_name))
+
+    exit_code, command_stdout, command_stderr = get_command_result(restart_command)
+    if exit_code == 0:
+        check_container_running(container_name)
```

Comment:
do we need this ? not sure if doing this is useful.

Reply:
I think the exit code of `0` indicates the command was executed successfully. But I want to check the container is actually restarted and in the running state.

Reply:
@jleveque Do you have any suggestion: should we check the container is running or not after executing the command to restart the container?

Reply:
i think this script job is to restart and not sure it should check if container is restarted or not. For ex: when we do `config reload` we just stop/start the service but do not check if container is running or not. In my opinion we should follow same approach.  

@jleveque what do you think ?

Reply:
If we had the container_checker in 201911, then it would take care of alerting us if the container didn't start up. I see your point that we should keep this simple. Maybe we can consider backporting the container_checker at some point?

Reply:
yes i think if we decide to backport  we can have this as part of the `container_check`  feature. @yozhao101 for now we can remove this . Also we have process monit so if telemetry container is not running then process not running monit syslog should be able to alert.

Also have you tested case when when we call  `check_memory_usage()` if telemetry container is not running what is happenig ?

Reply:
Great suggestion! The script `memory_checker` should check the memory usage if container was not running! 

If the container is not running, command `docker stats --no-stream --format {{.MemUsage}} <container_name>` will return `0B / 0B`.  This issue is fixed in the script.